### PR TITLE
Run backup commands in a shell

### PR DESCRIPTION
**Pull request by guewen**
The initial implementation of the backups do not allow to inject
environment variables in the commands. For instance, with the following
block:

    backup:
      command: echo
      args: "backup command on ${DB_NAME}"
      stop_on_failure: true
      ignore_if: test "${RUNNING_ENV}" != "prod"

ignore_if would really compare the "${RUNNING_ENV}" chain without
replacing it by the environment variable's value. Similarly, the DB_NAME
in the command args would be printed the exact same way without
interpolation.

Now, the operations can be wrapped in a shell (sh), which will take care
of the interpolation. The version operations are still run without
a shell (although you can use 'sh -c' in the yaml file if you need), but
now the backup commands use one.

This change has few effects on the existing behavior:

* pexpect.spawn can and is documented to use a string for the command +
args, so we no longer split them as a list, this is also safer as
splitting can be tricky in some cases
* as a result of no longer splitting the command, the output may change
a bit, for instance "echo 'foo bar'" was printed as "echo foo bar" in
the output, now it's "echo 'foo bar'"
* The backup options in the yaml file no longer expect command and args
in separate keys, only "command" is expected:

      backup:
        command: echo "backup command on ${DB_NAME}"
        stop_on_failure: true
        ignore_if: test "${RUNNING_ENV}" != "prod"

### CODE REVIEW

File: marabunta/model.py [hunk 1]
```diff
@@ -281,27 +279,30 @@ def operation(self, exclude_addons=None):
...
+        self.shell = shell
 
-    @staticmethod
-    def _shlex_split_unicode(command):
-        if sys.version_info < (3, 4):
-            return [l.decode('utf8') for l in shlex.split(
-                command.encode('utf-8'))]
+    def _spawn_command(self):
+        if self.shell:
+            return "sh -c '%s'" % (self.command,)
```
Comment (yvaucher) [inline marabunta/model.py]: What if `self.command` contains params with `'` ?

As for your example

`echo "backup command on ${DB_NAME}"`

What if I replace double quotes by single quotes

`echo 'backup command on ${DB_NAME}'`

Command will be incomplete.

```
└─> sh -c 'echo 'backup command on ${DB_NAME}''
backup
```

To escape char you will probably want to replace all `'` by `'"'"'` or `'\''` in self.command. (both use concatenation to close the quote, add an quote char and then reopen the quote) 

See https://stackoverflow.com/questions/1250079/how-to-escape-single-quotes-within-single-quoted-strings

This applies only to self.shell mode as there is no quote nesting.

Comment (guewen) [reply marabunta/model.py]: There is a better way :)

```python
>>> import pexpect
>>> import sys
>>> p = pexpect.spawn('sh', ['-c', 'echo "test test"'])
>>> p.logfile = sys.stdout
>>> p.expect(pexpect.EOF)
test test
0
>>> p.exitstatus
0
```

Comment (guewen) [reply marabunta/model.py]: Pushed as fixup commit

**Comment by guewen** [comment]
The environ of the tests is not helping me :-(

I think it's due to tox

```
    def test_shell_operation(capfd):
        # must be able to read env. variables
>       user = os.environ['USER']
tests/test_operation.py:47:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
self = environ({'PYTHONHASHSEED': '2361870612', 'LANG': 'en_US.UTF-8', 'PYTEST_CURRENT_TEST': 'tests/test_operation.py::test_...n:/usr/sbin:/usr/bin:/sbin:/bin:/home/travis/.rvm/bin:/home/travis/.phpenv/bin:/opt/pyenv/bin:/home/travis/.yarn/bin'})
```

**Comment by guewen** [comment]
@yvaucher fixed

**Comment by yvaucher** [comment]
squashed and merged

**Comment by guewen** [comment]
Thanks!

**Pull request merged**